src/component/Home.jsx

```diff
@@ -0,0 +1,373 @@
+import React, { useEffect, useState } from "react";
+import { Box, Button, Stack, Typography, Card } from "@mui/material";
+import { collection, deleteDoc, doc, getDocs } from "firebase/firestore";
+import { auth, database } from "../firebase";
+import { NavLink, useNavigate } from "react-router-dom";
+import Image from "../assets/flower1.jpg";
+import Navbar from "../component/Navbar";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+function Home() {
+  const [value, setValue] = useState([]);
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const getData = async () => {
+     
+      try {
+        const currentUser = auth.currentUser;
+        if (currentUser) {
+          const userId = currentUser.uid;
+          setLoading(true) 
+
+          const querySnapshot = await getDocs(
+            collection(database, `demo/${userId}/posts`)
+          );
+          const data = querySnapshot.docs.map((doc) => ({
+            id: doc.id,
+            ...doc.data(),
+          }));
+           
+
+          setValue(data);
+         setLoading(false);
+        }
+      } catch (error) {
+        console.error("Error getting documents: ", error);
+        setLoading(false);
+      }
+    };
+ 
+
+    getData();
+  }, []);
+
+  console.log(value, "value");
+
+  const handleDelete = async (id) => {
+    try {
+      const currentUser = auth.currentUser;
+      const userId = currentUser.uid;
+      const deletvalue = doc(database, `demo/${userId}/posts`, id);
+      await deleteDoc(deletvalue);
+      setValue((prevValue) => prevValue.filter((item) => item.id !== id));
+    } catch (error) {
+      console.error("Error deleting document: ", error);
+    }
+  };
+
+  const handleEdit = (id) => {
+    navigate(`/edit/${id}`);
+  };
+
+  return (
+    <>
+      {loading ? (
+        <h1>Loading...</h1>
```

Comment:
this loading should be visible in the center with a loading icon

Reply:
done